akka-persistence-typed/src/test/scala/akka/persistence/typed/scaladsl/PerformanceSpec.scala

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (C) 2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.persistence.typed.scaladsl
+
+import java.util.UUID
+
+import akka.actor.typed.scaladsl.Behaviors
+import akka.actor.typed.{ ActorRef, SupervisorStrategy, TypedAkkaSpecWithShutdown }
+import akka.persistence.typed.scaladsl.PersistentBehaviors.CommandHandler
+import akka.testkit.typed.TE
+import akka.testkit.typed.scaladsl.{ ActorTestKit, TestProbe }
+import com.typesafe.config.{ Config, ConfigFactory }
+import org.scalatest.concurrent.Eventually
+
+import scala.concurrent.duration._
+
+object PerformanceSpec {
+
+  val config =
+    """
+      akka.persistence.performance.cycles.load = 100
+      # more accurate throughput measurements
+      #akka.persistence.performance.cycles.load = 200000
+    """
+
+  sealed trait Command
+
+  case object StopMeasure extends Command
+
+  case class FailAt(sequence: Long) extends Command
+
+  case class CommandWithEvent(evt: String) extends Command
+
+  class Measure(numberOfMessages: Int) {
+    private val NanoToSecond = 1000.0 * 1000 * 1000
+
+    private var startTime: Long = 0L
+    private var stopTime: Long = 0L
+
+    def startMeasure(): Unit = {
+      startTime = System.nanoTime
+    }
+
+    def stopMeasure(): Double = {
+      stopTime = System.nanoTime
+      NanoToSecond * numberOfMessages / (stopTime - startTime)
+    }
+  }
+
+  case class Parameters(var persistCalls: Long = 0L, var failAt: Long = -1) {
+    def every(num: Long): Boolean = persistCalls % num == 0
+
+    def shouldFail: Boolean = persistCalls == failAt
+
+    def failureWasDefined: Boolean = failAt != -1L
+  }
+
+  def behavior(name: String, probe: TestProbe[Command])(other: (Command, Parameters) ⇒ Effect[String, String]) = {
+    Behaviors.supervise({
+      val parameters = Parameters()
+      PersistentBehaviors.receive[Command, String, String](
+        persistenceId = name,
+        "",
+        commandHandler = CommandHandler.command {
+          case StopMeasure      ⇒ Effect.none.andThen(probe.ref ! StopMeasure)
+          case FailAt(sequence) ⇒ Effect.none.andThen(_ ⇒ parameters.failAt = sequence)
+          case command          ⇒ other(command, parameters)
+        },
+        eventHandler = {
+          case (state, _) ⇒ state
+        }
+      ).onRecoveryCompleted {
+          case (_, _) ⇒ if (parameters.every(1000)) print("r")
+        }
+    }).onFailure(SupervisorStrategy.resume)
+  }
+
+  def eventSourcedTestPersistenceBehavior(name: String, probe: TestProbe[Command]) =
+    behavior(name, probe) {
```

Comment:
match may not be exhaustive.
[warn] It would fail on the following inputs: (FailAt(_), _), (StopMeasure, _)

Reply:
Fixed